lib/Sema/CSSimplify.cpp

```diff
@@ -7802,6 +7802,17 @@ ConstraintSystem::simplifyKeyPathApplicationConstraint(
     return SolutionKind::Unsolved;
   };
 
+  // For KeyPathDynamicMemberLookup a subscript with any label different
+  // than dynamicMember isn't applicable.
+  if (locator.getBaseLocator()->isForKeyPathDynamicMemberLookup()) {
+    if (auto *subscriptExpr =
+            dyn_cast_or_null<SubscriptExpr>(locator.getAnchor())) {
+      if (subscriptExpr->getArgumentLabels().size() == 1 &&
+          subscriptExpr->getArgumentLabels()[0].str() != "dynamicMember")
```

Comment:
When does the `dynamicMember:` argument label case occur? I think we just want to reject any attempt to use key path application through dynamic member lookup.

Reply:
You right, it doesn't occur. I thought when you use key path dynamic member lookup with a key path subscript it was going through the same path of a key path application, therefore the actual key path dynamic member key path subscript(`dynamicMember`) needed to be allowed... sorry :)

Reply:
Just fixed, thank you :)

Reply:
Thanks! Sorry, I have one more comment, but otherwise this looks good to me :)